docs/eventing/broker-trigger.md

```diff
@@ -0,0 +1,268 @@
+# Broker and Trigger
+
+Broker and Trigger are CRDs providing an event delivery mechanism that hides the
+details of event routing from the event producer and event consumer.
+
+## Broker
+
+Broker represents an 'event mesh'. Events are sent to the Broker's ingress and
+are then sent to any subscribers that are interested in that event. Once inside
+a Broker, all metadata other than the CloudEvent is stripped away (e.g. unless
+set as a CloudEvent attribute, there is no concept of how this event entered the
+Broker).
+
+Example:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: default
+spec:
+  channelTemplate:
+    provisioner:
+      apiVersion: eventing.knative.dev/v1alpha1
+      kind: ClusterChannelProvisioner
+      name: gcp-pubsub
+```
+
+## Trigger
+
+Trigger represents a desire to subscribe to events from a specific Broker. Basic
+filtering on the types of events is provided.
+
+Example:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: my-service-trigger
+spec:
+  filter:
+    sourceAndType:
+      type: dev.knative.foo.bar
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1alpha1
+      kind: Service
+      name: my-service
+```
+
+## Usage
+
+### ClusterChannelProvisioner
+
+`Broker`'s use their `spec.channelTemplate` to create their internal `Channel`s,
+which dictate the durability guarantees of events sent to that `Broker`. If
+`spec.channelTemplate` is not specified, then the
+[default provisioner](https://www.knative.dev/docs/eventing/channels/default-channels/)
+for their namespace is used.
+
+#### Setup
+
+Have a `ClusterChannelProvisioner` installed and set as the
+[default provisioner](https://www.knative.dev/docs/eventing/channels/default-channels/)
+for the namespace you are interested in. For development, the
+[`in-memory` `ClusterChannelProvisioner`](https://github.com/knative/eventing/tree/master/config/provisioners/in-memory-channel#deployment-steps)
+is normally used.
+
+#### Changing
+
+**Note** changing the `ClusterChannelProvisioner` of a running `Broker` will
+lose all in-flight events.
+
+If you want to change which `ClusterChannelProvisioner` is used by a given
+`Broker`, then determine if the `spec.channelTemplate` is specified or not.
+
+If `spec.channelTemplate` is specified:
+
+1. Delete the `Broker`.
+1. Create the `Broker` with the updated `spec.channelTemplate`.
+
+If `spec.channelTemplate` is not specified:
+
+1. Change the
+   [default provisioner](https://github.com/knative/docs/blob/master/docs/eventing/channels/default-channels.md#setting-the-default-channel-configuration)
+   for the namespace that `Broker` is in.
+1. Delete and recreate the `Broker`.
+
+### Broker
+
+There are two ways to create a Broker, via [namespace annotation](#annotation)
```

Comment:
Make it a list ?
---

There are two ways to create a Broker:
* via namespace annotation
* or manual setup